src/test/regress/expected/distributed_transactions.out

```diff
@@ -0,0 +1,1233 @@
+ Transaction with duplicate error in the local statement
+(1 row)
+
+BEGIN;
+INSERT INTO distributed_transaction_function.test_txn_dist VALUES (20);
+NOTICE:  issuing BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED;SELECT assign_distributed_transaction_id(xx, xx, 'xxxxxxx');
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+NOTICE:  issuing INSERT INTO distributed_transaction_function.test_txn_dist_900000 (intcol) VALUES (20)
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+-- This call will insert both the rows locally on the remote worker
+SELECT insert_data(21, 22);
+DEBUG:  pushing down function call in a multi-statement transaction
+DEBUG:  pushing down the function call
+NOTICE:  issuing BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED;SELECT assign_distributed_transaction_id(xx, xx, 'xxxxxxx');
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+NOTICE:  issuing SELECT distributed_transaction_function.insert_data(21, 22) AS insert_data
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+ insert_data
+---------------------------------------------------------------------
+
+(1 row)
+
+INSERT INTO distributed_transaction_function.test_txn_dist VALUES (23);
+NOTICE:  issuing INSERT INTO distributed_transaction_function.test_txn_dist_900028 (intcol) VALUES (23)
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+-- This will fail
+INSERT INTO distributed_transaction_function.test_txn_dist VALUES (23);
+NOTICE:  issuing INSERT INTO distributed_transaction_function.test_txn_dist_900028 (intcol) VALUES (23)
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+NOTICE:  issuing ROLLBACK
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+ERROR:  duplicate key value violates unique constraint "test_txn_dist_pkey_900028"
+DETAIL:  Key (intcol)=(23) already exists.
+CONTEXT:  while executing command on localhost:xxxxx
+COMMIT;
+SELECT 'Transaction with function doing remote connection' Testing;
+                      testing
+---------------------------------------------------------------------
+ Transaction with function doing remote connection
+(1 row)
+
+BEGIN;
+-- This statement will pass
+INSERT INTO distributed_transaction_function.test_txn_dist VALUES (30);
+NOTICE:  issuing BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED;SELECT assign_distributed_transaction_id(xx, xx, 'xxxxxxx');
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+NOTICE:  issuing INSERT INTO distributed_transaction_function.test_txn_dist_900028 (intcol) VALUES (30)
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+-- This call will insert both one rows locally and another on a different node
+SELECT insert_data(31, 33);
+DEBUG:  pushing down function call in a multi-statement transaction
+DEBUG:  pushing down the function call
+NOTICE:  issuing SELECT distributed_transaction_function.insert_data(31, 33) AS insert_data
```

Comment:
should fail since we're inserting 33 when the partition key value is 31.

Reply:
Yes, same as above.
![image](https://user-images.githubusercontent.com/44680808/141184519-b03d21ff-23d3-4167-8c3b-ec1bad6885bb.png)